File: 250121/점 개수 세기 3/count-number-of-points-3.cpp
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;

int main() {
    // Please write your code here.
    int arr[100001] = {};

    int n, q;
    scanf("%d %d", &n, &q);
    for (int i=0; i<n; i++)  
        scanf("%d", &arr[i]);

    sort(arr, arr+n);

    map<int, int> mapper;
    for (int i=0; i<n; i++) {
        mapper[arr[i]] = i+1;
        arr[i] = mapper[arr[i]];
    }
    
    int arr2[100001] = {};
    for (int i=0; i<n; i++) {
        arr2[arr[i]] = 1;
    }
    
    int prefix[100001] = {};
    for (int i=1; i<=n; i++) {
        prefix[i] = arr2[i] + prefix[i-1];
    }

    for (int i=0; i<q; i++) {
        int a, b;
        scanf("%d %d", &a, &b);
        a = mapper[a];
        b = mapper[b];
        cout << prefix[b] - prefix[a-1] << '\n';
    }
    return 0;
}

File: 250701/점 개수 세기 3/count-number-of-points-3.py
N, Q = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()

mapper = {}
for i in range(N):
    mapper[arr[i]] = i+1

for i in range(N):
    arr[i] = mapper[arr[i]]

arrr = [0 for i in range(100010)]
for i in range(N):
        arrr[arr[i]] = 1

prefix = [0 for i in range(100010)]
for i in range(1, 100010):
    prefix[i] = arrr[i] + prefix[i-1]

for q in range(Q):
    a, b = map(int, input().split())
    print(prefix[mapper[b]] - prefix[mapper[a]-1])